docs: add metadata to all remaining pages (#1246)
canonical/rockcraft: docs/reference/extensions/django-framework.rst

.. meta::

>     :description: Reference documentation for the Rockcraft Django framework extension, which configures Django rocks with Gunicorn and installs dependencies at /django/app.

Reviewer: ```suggestion
    :description: Reference documentation for the Django framework extension, which configures Django in a rock and populates it with Django dependencies such as Gunicorn.
```

Let's try this format for extensions. This is actually the same format in the [examples](https://documentation.ubuntu.com/starflow/latest/how-to/add-a-page-meta-description/#examples).
